packages/js/src/initializers/estimated-reading-time.js

```diff
@@ -41,42 +42,41 @@ function initializeEstimatedReadingTimeClassic() {
 	} );
 }
 
+// Used to trigger the initial reading time calculation for the block and Elementor editors.
+let previousContent = "";
+
 /**
- * Initializes the estimated reading time for the block editor.
+ * Gets the estimated reading time in the block editor if the content has changed.
  *
  * @returns {void}
  */
-function initializeEstimatedReadingTimeBlockEditor() {
-	let previousContent = select( "core/editor" ).getEditedPostAttribute( "content" );
-
-	subscribe( () => {
-		const content = select( "core/editor" ).getEditedPostAttribute( "content" );
-
-		if ( content !== previousContent ) {
-			previousContent = content;
-			debouncedGetEstimatedReadingTime( content );
-		}
-	} );
+function getEstimatedReadingTimeBlockEditor() {
+	const content = select( "core/editor" ).getEditedPostAttribute( "content" );
+	if ( previousContent !== content ) {
+		previousContent = content;
+		getEstimatedReadingTime( content );
+	}
 }
 
+// Delays execution by 1,5 seconds for any change, forces execution after 3 seconds.
+const debounceBlockEditor = debounce( getEstimatedReadingTimeBlockEditor, 1500, { maxWait: 3000 } );
```

Comment:
This should be called `debouncedGetEstimatedReadingTimeBlockEditor`, the name implies it applies the debouncing itself. If you choose to apply my feedback below this issue can be avoided because no separate variable will be created for this function.

Reply:
Good point :)